frontend/src/features/material-details/components/StockModal.tsx

```diff
@@ -0,0 +1,369 @@
+import { GridItem } from '@components/ui/GridItem';
+import { useSites } from '@features/stock-view/hooks/useSites';
+import { useNotifications } from '@contexts/notificationContext';
+import { deleteStocks, postStocks } from '@services/stocks-service';
+
+const createStockColumns = (handleDelete?: (row: Stock) => void) => {
+    const columns = [
+        {
+            field: 'quantity',
+            headerName: 'Quantity',
+            headerAlign: 'center',
+            flex: 1.2,
+            renderCell: (data: { row: Stock }) => (
+                <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                    {`${data.row.quantity} ${getUnitOfMeasurement(data.row.measurementUnit)}`}
+                </Box>
+            ),
+        },
+        {
+          field: 'stockLocationBpns',
+          headerName: 'Site',
+          headerAlign: 'center',
+          flex: 2,
+          renderCell: (data: { row: Stock }) => (
+              <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                  {data.row.stockLocationBpns}
+              </Box>
+          ),
+      },
+        {
+            field: 'partner',
+            headerName: 'Partner',
+            headerAlign: 'center',
+            flex: 2,
+            renderCell: (data: { row: Stock }) => (
+                <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                    {data.row.partner?.name}
+                </Box>
+            ),
+        },
+        {
+            field: 'lastUpdatedOn',
+            headerName: 'Updated',
+            headerAlign: 'center',
+            flex: 1.5,
+            renderCell: (data: { row: Stock }) => (
+                <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                    {new Date(data.row.lastUpdatedOn).toLocaleTimeString('en-GB')}
+                </Box>
+            ),
+        },
+        {
+            field: 'customerOrderNumber',
+            headerName: 'Order Reference',
+            headerAlign: 'center',
+            flex: 2,
+            renderCell: (data: { row: Stock }) => (
+                <Box display="flex" textAlign="center" alignItems="center" justifyContent="center" width="100%" height="100%">
+                    {data.row.customerOrderNumber ? (
+                        <Stack>
+                            <Box>{`${data.row.customerOrderNumber} / ${data.row.customerOrderPositionNumber}  `}</Box>
+                            <Box>{data.row.supplierOrderNumber || '-'}</Box>
+                        </Stack>
+                    ) : (
+                        '-'
+                    )}
+                </Box>
+            ),
+        },
+        {
+            field: 'isBlocked',
+            headerName: 'is blocked',
```

Comment:
```suggestion
            headerName: 'Is Blocked',
```

Please write uppercase. Maybe "Blocked" is even better.